tensorboard/plugins/mesh_visualizer/tf_mesh_dashboard/mesh-loader.js

```diff
@@ -0,0 +1,287 @@
+     */
+    selectedView: {type: String, value: 'all'},
+    /** @type {!Object} Request manager to communicate with the server. */
+    requestManager: Object,
+    /**
+     * @type {!Object} Component to render meshes and point
+     * clouds.
+     */
+    _meshViewer: {type: Object},
+    /**
+     * @type {!Object} Data provider to
+     * communicate with the server and parse raw data.
+     */
+    _dataProvider: {type: Object},
+    /**@type {!Object} Wrapper to function to transform value into color.*/
+    _colorScaleFunction: {
+      type: Object,  // function: string => string
+      value: () => tf_color_scale.runsColorScale,
+    },
+    /**@type {string} Wrapper around function to compute color for a run.*/
+    _runColor: {
+      type: String,
+      computed: '_computeRunColor(run)',
+    },
+    /** @type {!Array} Contains datums for each step. */
+    _steps: {
+      type: Array,
+      value: () => [],
+      notify: true,
+    },
+    /** @type {number} Contains index of current step. */
+    _stepIndex: {
+      type: Number,
+      notify: true,
+    },
+    /** @type {!Object} Contains datum of current step. */
+    _currentStep: {
+      type: Object,
+      computed: '_computeCurrentStep(_steps, _stepIndex)',
+    },
+    /** @type {!bool} Determines if mesh viewer attached to dom. */
+    _meshViewerAttached: {type: Boolean, value: false},
+    /**
+     * @type {!bool} Determines if camera position was initially set to
+     * defaults.
+     */
+    _cameraPositionInitialized: {type: Boolean, value: false},
+    /** @type {!bool} Determines if at least one step is present. */
+    _hasAtLeastOneStep: {
+      type: Boolean,
+      computed: '_computeHasAtLeastOneStep(_steps)',
+    },
+    /** @type {!bool} Determines if multiple steps are present. */
+    _hasMultipleSteps: {
+      type: Boolean,
+      computed: '_computeHasMultipleSteps(_steps)',
+    },
+    /**
+     * @type {number} Contains current step value (step number assigned
+     * during training).
+     */
+    _stepValue: {
+      type: Number,
+      computed: '_computeStepValue(_currentStep)',
+    },
+    /** @type {string} Contains formatted wall time. */
+    _currentWallTime: {
+      type: String,
+      computed: '_computeCurrentWallTime(_currentStep)',
+    },
+    /** @type {number} Maximum step index loaded to the client. */
+    _maxStepIndex: {
+      type: Number,
+      computed: '_computeMaxStepIndex(_steps)',
+    },
+    /** @type {string} Formatted sample string. */
+    _sampleText: {
+      type: String,
+      computed: '_computeSampleText(sample)',
+    },
+    /** @type {!bool} Defines if multiple samples are present. */
+    _hasMultipleSamples: {
+      type: Boolean,
+      computed: '_computeHasMultipleSamples(ofSamples)',
+    },
+    /** @type {!bool} Defines if browser still loading data. */
+    _isMeshLoading: {
+      type: Boolean,
+      value: false,
+    }
+  },
+
+  observers: [
+    'reload(run, tag)',
```

Comment:
Can we introduce new property called `active` and decide whether to really reload based on that? In Polymer2 (as opposed to Polymer1 we are currently using), your current way of handling `active`ness is not ideal. Thanks!

Reply:
How this property will be set? By what component? Do you have an example somewhere?

Reply:
It would be set by mesh-dashboard. It can be hard coded to `true` for now though.